controllers/topology/desired_state.go

```diff
@@ -236,6 +236,11 @@ func computeMachineDeployment(_ context.Context, s *scope.Scope, machineDeployme
 		currentObjectRef:      currentBootstrapTemplateRef,
 	})
 
+	// Add ClusterTopologyMachineDeploymentLabel to the generated Bootstrap template
+	bootstrapTemplateLabels := map[string]string{}
```

Comment:
Let's keep the labels from desiredMachineDeployment and initialise them to empty map only if nil

```suggestion
	bootstrapTemplateLabels := desiredMachineDeployment.BootstrapTemplate.GetLabels()
	if  bootstrapTemplateLabels  == nil {
		bootstrapTemplateLabels = map[string]string{}
	}
```